aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -1182,6 +1188,7 @@ public void onFailure(IMqttToken asyncActionToken, Throwable e) {
                         connectionState = MqttManagerConnectionState.Disconnected;
                         userConnectionCallback(e);
                     }
+                    sessionPresent = asyncActionToken.getSessionPresent();
                 }
             });
         } catch (final MqttException e) {
```

Comment:
Would it make sense to also update `sessionPresent` here?  It appears that `mqttClient.connect(...)` can throw an `MqttException`, but then still just recover, and call the `userConnectionCallback()`